pasqal-cloud/pasqal_cloud/__init__.py

```diff
@@ -147,17 +149,31 @@ def __init__(
             endpoints: Endpoints targeted of the public apis.
             auth0: Auth0Config object to define the auth0 tenant to target.
             project_id: ID of the owner project of the batch.
+            client_class: client class used to determine authentification \
+            and endpoints targeted by the SDK
         """
         _check_sdk_version()
 
-        self._client = Client(
-            project_id=project_id,
-            username=username,
-            password=password,
-            token_provider=token_provider,
-            endpoints=endpoints,
-            auth0=auth0,
-        )
+        # Collect all arguments we want to pass
+        # to instantiate a Client
+        client_kwargs_arguments = {
+            "username": username,
+            "password": password,
+            "token_provider": token_provider,
+            "endpoints": endpoints,
+            "auth0": auth0,
+            "project_id": project_id,
+        }
+
+        # Filter only what the client_class.__init__ accepts
+        client_signature = inspect.signature(client_class.__init__)
+        accepted = {
+            k: v
+            for k, v in client_kwargs_arguments.items()
+            if k in client_signature.parameters and v is not None
+        }
+
+        self._client = client_class(**accepted)  # type: ignore[arg-type]
```

Comment:
If this is what you need to get rid of the argument in the constructor of the OVHClient I propose we keep them instead.

Maybe another suggestion is just to have it defined as:

    def __init__(
        self,
        token_provider: Optional[TokenProvider] = None,
        **kwargs
    ):
        super().__init__(
            token_provider=token_provider,
            **kwargs
        )